platform/engineering/request-for-change/2019-09-20-rfc-cms-based-alerts.md

```diff
@@ -0,0 +1,84 @@
+1. **Teamsite**. This is going away once legacy VA.gov pages are migrated to WBC VA.gov or retired.
+1. **Sourced from `va.gov-content` on load**. Requires content managers to be comfortable with GitHub and markdown or YAML.
+1. **Sourced from Vets API on load.** No logical connection between the CMS and Vets API exists.
+
+## Possible design approaches
+
+For the first two possibilities, assume that a change to a specific area of the content model causes the CMS to tell Jenkins kick off a specific pipeline with certain parameters. For the third, Jenkins is out of the loop.
+
+### Single page BRD
+
+Jenkins makes a GraphQL query the CMS content API based on given parameters, pulls _only_ the relevant content and flags, builds _only_ the corresponding templates, runs _only_ the relevant tests and checks, and uploads the HTML asset to the existing S3 bucket, overwriting the existing asset. As soon as upload is complete, the page begins being served to users. Alerts are rendered along with the rest of the static content on the page.
+
+|Pros|Cons|
+|----|----|
+|Familiar architecture and process|Somewhat weighty process for minor updates|
+|Can apply existing tests (e.g. link checker, 508)|Inherently slower b/c of tests and checks|
+|No additional client requests needed after page download|
+|Leverages benefits of GraphQL|
+
+### Sourced from S3 bucket on load
+
+Jenkins makes a GraphQL query to the CMS content API based on given parameters, pulls _only_ the relevant content and flags, builds one or more JSON, YAML, or other type of structured data file, and uploads to a dedicated S3 bucket, possibly overwriting the existing asset. On page load, a React component makes an asynchronous call to the S3 bucket, parses the payload for any relevant alert information, and renders the result (if applicable).
+
+|Pros|Cons|
+|----|----|
+|Lightweight and fast|Need to track another kind of asset outside of source control|
+|Leverages benefits of GraphQL|Adds more architectural and process complexity website management|
+| |Requires at least one more client request after page download|
+
+### Sourced from CMS content API on load
+
+A React component makes an asynchronous call with page-specific parameters to api.va.gov (or some other middleware that shares network or environment access with the CMS content API) which, in turn, makes a GraphQL call to the CMS content API for _only_ the relevant content and flags. The payload is returned to the React component which renders the result (if applicable).
+
+|Pros|Cons|
+|----|----|
+|Supports decoupled approach|Big departure from current architecture and process|
+|Removes website/CMS interdependency if extended to all CMS data|Requires at least one more client request after page download
+|Allows generic content to be easily consumed by other applications|Creates new threat vector for the CMS|
+|Doesn't require tracking of additional asset|
+|Leverages benefits of GraphQL||
```

Comment:
A con here: the build process for vets-website already includes an implementation to pull from the CMS, we'd have to duplicate code in order to make this work via the API.